[DOCS] "Create rule" updates: rule preview, saved queries (#2559)
elastic/security-docs: docs/detections/rules-ui-create.asciidoc

{ref}/getting-started-index.html[Index some documents],
{ref}/indices-create-index.html[Create index API], and
{ref}/mapping-types.html[Field data types]. If you have indicators in a standard file format, such as CSV or JSON, you can also use the Machine Learning Data Visualizer to import your indicators into an indicator index. See {ml-docs}/ml-getting-started.html#sample-data-visualizer[Explore the data in {kib}] and use the *Import Data* option to import your indicators.
+
TIP: You can also use value lists as the indicator match index. See <<indicator-value-lists>> at the end of this topic for more information.

* <<create-new-terms-rule, *New terms*>>: Generates an alert for each new term detected in source documents within a specified time range.

> When modifying rules or managing detection alerts, you can <<detections-ui-exceptions,add exceptions>> that prevent a rule from generating an alert even when its criteria are met. This is useful for reducing noise, such as preventing alerts from trusted processes and internal IP addresses.

Author: I agree with changing to plural, but not sure about "such as by" — maybe it's technically correct grammar but feels less natural
```suggestion
When modifying rules or managing detection alerts, you can <<detections-ui-exceptions,add exceptions>> that prevent a rule from generating alerts even when its criteria are met. This is useful for reducing noise, such as preventing alerts from trusted processes and internal IP addresses.
```

.. Use the filter and query fields to create the criteria used for detecting
alerts.
+
The following example (based on the prebuilt rule <<prebuilt-rule-0-14-2-volume-shadow-copy-deleted-or-resized-via-vssadmin, Volume Shadow Copy Deleted or Resized via VssAdmin>>) detects when the `vssadmin delete shadows`
Windows command is executed:

** *Index patterns*: `winlogbeat-*`
+
Winlogbeat ships Windows event logs to {elastic-sec}.

** *Custom query*: `event.action:"Process Create (rule: ProcessCreate)" and process.name:"vssadmin.exe" and process.args:("delete" and "shadows")`
+
Searches the `winlogbeat-*` indices for `vssadmin.exe` executions with
the `delete` and `shadow` arguments, which are used to delete a volume's shadow
copies.
+
[role="screenshot"]
image::images/rule-query-example.png[Rule query example]

.. You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,18,18]) and queries from saved Timelines (*Import query from saved Timeline*) as rule conditions.
+

> When you use a saved query, the *Load saved query "_query name_" dynamically on each rule execution* check box appears:

Reviewer: is this worth to mention changes on rule details page for saved query rules?

https://github.com/elastic/kibana/issues/139250#issuecomment-1247918828

Author: Hi @vitaliidm I think it's worth mentioning, but maybe not in this section. The reader is in the middle of a kind of long procedure of creating a rule, so they won't be able to observe the details page until they're done with all the steps and save the rule.

Currently we actually don't have a dedicated docs page for the rule details page, but we're planning to create one for reasons just like this. I've added a [comment](https://github.com/elastic/security-docs/issues/2560#issuecomment-1278324186) to that docs issue so we can be sure to include this detail.

* Select this to use the saved query every time the rule runs. You won't be able to modify the rule's *Custom query* field or filters, because the rule will only use the saved query's settings. To make changes, modify the saved query itself.

> * Deselect this to load the saved query as a one-time way of populating the rule's *Custom query* field and filters. You can then make further adjustments to the rule's query and filters as needed. If the saved query is later changed, the rule will not inherit those changes.

Reviewer 2: It might be useful to describe the first option as a "reference" or "link" to the saved query, and the second as a "copy" of the query and filters from the saved query.

Author: ```suggestion
* Select this to use the saved query every time the rule runs. This links the rule to the saved query, and you won't be able to modify the rule's *Custom query* field or filters because the rule will only use settings from the saved query. To make changes, modify the saved query itself.

* Deselect this to load the saved query as a one-time way of populating the rule's *Custom query* field and filters. This merely copies the settings from the saved query to the rule, so you can then make further adjustments to the rule's query and filters as needed. If the saved query is later changed, the rule will not inherit those changes.
```

. Click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.

[discrete]
[[preview-rules]]
=== Preview your rule (optional)

You can preview any rule before saving it to learn how noisy it will be. If necessary, you can then adjust the rule's query or other settings.

NOTE: To preview rules, you need the `read` privilege to the `.preview.alerts-security.alerts-<space-id>` index and `All` privileges for the Security feature. Refer to <<detections-permissions-section>> for more information.

Click the *Rule preview* button while creating or editing a rule. The preview opens in a side panel, showing a histogram and table with the alerts you can expect, based on the defined rule settings and past events in your indices. 

[role="screenshot"]
image::images/preview-rule.png[Rule preview]

The preview also includes the effects of rule exceptions and override fields. In the histogram, alerts are stacked by `event.category` (or `host.name` for machine learning rules), and events with multiple values are counted more than once.

To interact with the rule preview:

* Use the date and time picker to define the preview's time range.
+

> TIP: Avoid setting long time ranges with short rule intervals. This might cause the rule preview to timeout.

Reviewer 3: ```suggestion
TIP: Avoid setting long time ranges with short rule intervals, or the rule preview might time out.
```

+
For example, if a rule has an interval of 5 minutes, no additional look-back time, and a history window size of 7 days, a term will be considered new only if the time it appears within the last 7 days is also within the last 5 minutes. Configure the rule interval and additional look-back time when you <<rule-schedule, set the rule's schedule>>.
. Click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.

[discrete]
[[preview-rules]]
=== Preview your rule (optional)

You can preview any custom or prebuilt rule to find out how noisy it will be. For a custom rule, you can then adjust the rule's query or other settings.

NOTE: To preview rules, you need the `read` privilege to the `.preview.alerts-security.alerts-<space-id>` index and `All` privileges for the Security feature. Refer to <<detections-permissions-section>> for more information.

Click the *Rule preview* button while creating or editing a rule. The preview opens in a side panel, showing a histogram and table with the alerts you can expect, based on the defined rule settings and past events in your indices. 

[role="screenshot"]
image::images/preview-rule.png[Rule preview]

> The preview also includes the effects of rule exceptions and override fields. In the histogram, alerts are stacked by `event.category` (or `host.name` for machine learning rules), and events with multiple values are counted more than once.

Reviewer 2: ```suggestion
The preview also includes the effects of rule exceptions and override fields. In the histogram, alerts are stacked by `event.category` (or `host.name` for machine learning rules), and alerts with multiple values are counted more than once.
```

To interact with the rule preview:

* Use the date and time picker to define the preview's time range.
+
TIP: Avoid setting long time ranges with short rule intervals, or the rule preview might time out.

* Click *Refresh* to update the preview. 

> ** When you edit the rule's settings or the preview's time range, the button changes from blue (image:images/rule-preview-refresh-circle.png[Blue circular refresh icon,16,17]) to green (image:images/rule-preview-refresh-arrow.png[Green right-pointing arrow refresh icon,17,17]) to indicate that the edits might affect the preview.

Reviewer 2: Using "might affect" here sounds to me like we're not sure if the edits get included in the preview or not. Would something along the lines of "The button changes from blue to green to indicate that the rule has been edited since the last preview" work here?

Author: ```suggestion
** When you edit the rule's settings or the preview's time range, the button changes from blue (image:images/rule-preview-refresh-circle.png[Blue circular refresh icon,16,17]) to green (image:images/rule-preview-refresh-arrow.png[Green right-pointing arrow refresh icon,17,17]) to indicate that the rule has been edited since the last preview.
```
